parse_contributors_oo.py

```diff
@@ -0,0 +1,313 @@
+        ----------
+        user_data : dict
+            All contributor data from the website
+        update_keys : list
+            List of string values representing the key value pair items to
+            be updates
+        """
+
+        for i, item in enumerate(contrib_data):
+            gh_name = item["github_username"]
+            # Update the key:value pairs for data pulled from GitHub
+            # Note that some data needs to be manual updated such as which
+            # packages someone has reviewed. Cut if our team updates
+            # the google doc or another file maybe i can automate that too?
+            # TODO: I could parse github issues for reviewer names? and author names??!
+            for akey in update_keys:
+                # Stupid that the gh name is there twice??
+                item[akey] = gh_data[gh_name][gh_name][akey]
+        return contrib_data
+
+    def format_url(self, url: str) -> str:
+        """Append https to the beginning of URL if it doesn't exist
+        If the url doesn't have https add it
+        If the url starts with http change it to https
+        Else do nothing
+
+        Parameters
+        ----------
+        url : str
+            String representing the url grabbed from the GH api
+
+        """
+        if not url:
+            return url  # returns empty string if url is empty
+        elif url.startswith("https://"):
+            return url
+        elif url.startswith("http://"):
+            return "https://" + url[7:]
+        else:
+            return "https://" + url
+
+    def create_new_contrib_file(self, filename: str, contrib_data: dict):
+        """Update website contrib file with the information grabbed from GitHub
+        API
+
+        Serialize contrib data from dictionary to YAML file.
+
+        Parameters
+        ----------
+
+        filename : str
+            Name of the output contributor filename ().yml format)
+        contrib_data :  dict
+            A dict containing contributor data for the website.
+
+        Returns
+        -------
+        """
+
+        with open(filename, "w") as file:
+            # Create YAML object with RoundTripDumper to keep key order intact
+            yaml = ruamel.yaml.YAML(typ="rt")
+            # Set the indent parameter to 2 for the yaml output
+            yaml.indent(mapping=4, sequence=4, offset=2)
+            yaml.dump(contrib_data, file)
+
+    def clean_yaml_file(self, filename):
+        """Open a yaml file and remove extra indent and spacing"""
+        with open(filename, "r") as f:
+            lines = f.readlines()
+
+        cleaned_lines = []
+        for line in lines:
+            if line.startswith("  "):
+                cleaned_lines.append(line[2:])
+            else:
+                cleaned_lines.append(line)
+
+        cleaned_text = "".join(cleaned_lines).replace("''", "")
+
+        with open(filename, "w") as f:
+            f.write(cleaned_text)
+
+
+"""Begin actual script """
```

Comment:
not sure if you really need this code in the global scope of the module, 
but normally we create a function called `main` and we add everything there
and we call it at the end of the file with:

```
if __name__ = "__main__":
    main()
```